diff -Naur vtkcontrib_orig/src/tests/CMakeLists.txt vtkcontrib_dest/src/tests/CMakeLists.txt
--- vtkcontrib_orig/src/tests/CMakeLists.txt	2019-12-09 14:28:38.957843000 +0100
+++ vtkcontrib_dest/src/tests/CMakeLists.txt	2021-03-25 15:34:44.939082963 +0100
@@ -8,10 +8,15 @@
 target_include_directories (point_widget PRIVATE ../VtkContrib/public ${CMAKE_CURRENT_SOURCE_DIR})
 target_compile_options (point_widget PRIVATE ${SHARED_CFLAGS}) # Requested by Qt ...
 target_link_libraries (point_widget PUBLIC VtkContrib)
+
 target_include_directories (trihedrons PRIVATE ../VtkContrib/public ${CMAKE_CURRENT_SOURCE_DIR})
 target_compile_options (trihedrons PRIVATE ${SHARED_CFLAGS}) # Requested by Qt ...
 target_link_libraries (trihedrons PUBLIC VtkContrib)
 
+# for libmount
+target_link_libraries (point_widget PUBLIC intl)
+target_link_libraries (trihedrons PUBLIC intl)
+
 # INSTALL_RPATH modifie le rpath pour les libs internes au projet :
 set_target_properties (point_widget PROPERTIES INSTALL_RPATH_USE_LINK_PATH 1 INSTALL_RPATH ${CMAKE_PACKAGE_RPATH_DIR})
 set_target_properties (trihedrons PROPERTIES INSTALL_RPATH_USE_LINK_PATH 1 INSTALL_RPATH ${CMAKE_PACKAGE_RPATH_DIR})
diff -Naur vtkcontrib_orig/src/VtkContrib/CMakeLists.txt vtkcontrib_dest/src/VtkContrib/CMakeLists.txt
--- vtkcontrib_orig/src/VtkContrib/CMakeLists.txt	2021-02-01 08:42:31.149762000 +0100
+++ vtkcontrib_dest/src/VtkContrib/CMakeLists.txt	2021-03-25 14:59:40.892375454 +0100
@@ -57,6 +57,13 @@
 	set (VTK_MPI_CFLAGS "${ompi_CFLAGS}" "${ompi_CFLAGS_OTHER}")
 	set (VTK_MPI_LDFLAGS "${ompi_LDFLAGS}" "${ompi_LDFLAGS_OTHER}")
 	set (VTK_MPI_LIBS "${ompi_LINK_LIBRARIES}")
+elseif (${PLATFORM} STREQUAL "Rhel_8__x86_64")
+        include (${CMAKE_SOURCE_DIR}/cmake/ompi.cmake)
+        set (VTK_MPI_LIB_DIR "${ompi_LIBRARY_DIRS}")
+        set (VTK_MPI_INC_DIR "${ompi_INCLUDE_DIRS}")
+        set (VTK_MPI_CFLAGS "${ompi_CFLAGS}" "${ompi_CFLAGS_OTHER}")
+        set (VTK_MPI_LDFLAGS "${ompi_LDFLAGS}" "${ompi_LDFLAGS_OTHER}")
+        set (VTK_MPI_LIBS "${ompi_LINK_LIBRARIES}")
 elseif (${PLATFORM} STREQUAL "CentOS")
 	include (${CMAKE_SOURCE_DIR}/cmake/mpich.cmake)
 	set (VTK_MPI_LIB_DIR "${mpich_LIBRARY_DIRS}")
