diff -Naur VTK-7.1.1_orig/CMake/VTKGenerateExportHeader.cmake VTK-7.1.1_dest/CMake/VTKGenerateExportHeader.cmake
--- VTK-7.1.1_orig/CMake/VTKGenerateExportHeader.cmake	2017-03-20 16:26:17.000000000 +0100
+++ VTK-7.1.1_dest/CMake/VTKGenerateExportHeader.cmake	2022-12-12 11:35:23.181784029 +0100
@@ -214,8 +214,7 @@
     check_cxx_compiler_flag(-fvisibility=hidden COMPILER_HAS_HIDDEN_VISIBILITY)
     check_cxx_compiler_flag(-fvisibility-inlines-hidden
       COMPILER_HAS_HIDDEN_INLINE_VISIBILITY)
-    option(USE_COMPILER_HIDDEN_VISIBILITY
-      "Use HIDDEN visibility support if available." ON)
+    option(USE_COMPILER_HIDDEN_VISIBILITY "Use HIDDEN visibility support if available." ON)
     mark_as_advanced(USE_COMPILER_HIDDEN_VISIBILITY)
   endif()
 endmacro()
@@ -375,11 +374,11 @@
     return()
   endif()
 
-  set (EXTRA_FLAGS "-fvisibility=hidden")
+ # set (EXTRA_FLAGS "-fvisibility=hidden")	# CP comments in order to avoid undefined symbol (seems inefficient, use vtkModuleMacros.cmake patch)
 
-  if(COMPILER_HAS_HIDDEN_INLINE_VISIBILITY)
-    set (EXTRA_FLAGS "${EXTRA_FLAGS} -fvisibility-inlines-hidden")
-  endif()
+ # if(COMPILER_HAS_HIDDEN_INLINE_VISIBILITY)
+ #   set (EXTRA_FLAGS "${EXTRA_FLAGS} -fvisibility-inlines-hidden")	# CP comments in order to avoid undefined symbol (seems inefficient, use vtkModuleMacros.cmake patch)
+ # endif()
 
   # Either return the extra flags needed in the supplied argument, or to the
   # CMAKE_CXX_FLAGS if no argument is supplied.
diff -Naur VTK-7.1.1_orig/CMake/vtkModuleMacros.cmake VTK-7.1.1_dest/CMake/vtkModuleMacros.cmake
--- VTK-7.1.1_orig/CMake/vtkModuleMacros.cmake	2017-03-20 16:26:17.000000000 +0100
+++ VTK-7.1.1_dest/CMake/vtkModuleMacros.cmake	2022-12-12 11:36:32.708188423 +0100
@@ -727,8 +727,7 @@
       set_property(TARGET ${vtk-module}${target_suffix} APPEND
         PROPERTY COMPILE_FLAGS "${my_abi_flags}")
     else()
-      set_property(TARGET ${vtk-module}${target_suffix}
-        PROPERTY CXX_VISIBILITY_PRESET "hidden")
+#      set_property(TARGET ${vtk-module}${target_suffix}  PROPERTY CXX_VISIBILITY_PRESET "hidden")	# CP comments in order to avoid undefined symbol (pb with g++ 10/11, https://bugs.gentoo.org/723374)
     endif()
   endif()
 
