diff -ur lima-ref/src/Lima/cmake/LimaConfig.cmake.in lima/src/Lima/cmake/LimaConfig.cmake.in
--- lima-ref/src/Lima/cmake/LimaConfig.cmake.in	2020-10-12 14:08:26.888478742 +0200
+++ lima/src/Lima/cmake/LimaConfig.cmake.in	2020-10-12 14:21:13.117411669 +0200
@@ -5,6 +5,7 @@
 include(CMakeFindDependencyMacro)
 include (${CMAKE_CURRENT_LIST_DIR}/FindHdf145.cmake)
 find_dependency (HDF5)
+find_dependency (Threads)
 
 # EXPURGE_BEGINNING_TAG LOCAL_DEPENDENCIES
 set (MachineTypes_FOUND "@MachineTypes_ENABLED@")
